linera-core/src/node.rs

```diff
@@ -102,6 +102,24 @@ pub trait ValidatorNode {
     // certificate using this blob.
     async fn upload_blob(&self, content: BlobContent) -> Result<BlobId, NodeError>;
 
+    /// Uploads the blobs to the validator.
+    // Unfortunately, this doesn't compile as an async function: async functions in traits
+    // don't play well with default implementations, apparently.
+    // See also https://github.com/rust-lang/impl-trait-utils/issues/17
+    fn upload_blobs(
+        &self,
+        blobs: Vec<Blob>,
+    ) -> impl futures::Future<Output = Result<(), NodeError>> {
+        let tasks: Vec<_> = blobs
+            .into_iter()
+            .map(|blob| self.upload_blob(blob.into()))
+            .collect();
+        async move {
+            futures::future::try_join_all(tasks).await?;
+            Ok(())
+        }
```

Comment:
Isn't this equivalent:

```suggestion
        futures::future::try_join_all(tasks)
```

Reply:
Not quite, because this will return a `Result<Vec<BlobId>, NodeError>` instead of `Result<(), NodeError>`, but it doesn't hurt and is cleaner. Nice catch, thanks!